### PR TITLE
Rewrite and adapt to Scala 3 features

### PR DESCRIPTION
**Pull request by hamzaremmal**

### CODE REVIEW

File: src/main/scala/dotty/xml/interpolator/internal/Macro.scala [hunk 1]
```diff
@@ -53,48 +58,42 @@ object Macro {
...
+    import Expand.apply as expand
 
-    val interpolate = (
-      parse
-        andThen transform
-        andThen validate
-        andThen typecheck
-        andThen expand
-    )
+    val interpolate = parse andThen
```
Comment (nicolasstucki) [inline src/main/scala/dotty/xml/interpolator/internal/Macro.scala:71]:
```suggestion
    val interpolate = 
      parse andThen 
```

File: src/main/scala/dotty/xml/interpolator/internal/package.scala [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+  def apply(pos: Position): Int =
+    pos match
       case OffsetPosition(_, offset) => offset
       case _ => throw new Exception("expected offset position")
-    }
-  }
-}
+
+private[internal] inline def ctx(using XmlContext): XmlContext = summon
+private[internal] inline def reporter(using Reporter) : Reporter = summon
```
Comment (nicolasstucki) [inline src/main/scala/dotty/xml/interpolator/internal/package.scala:13]:
```suggestion
private[internal] inline def reporter(using Reporter): Reporter = summon

```

File: src/test/scala/dotty/xml/interpolator/TrailingWhitespaceTest.scala [hunk 3]
```diff
@@ -29,6 +30,7 @@ class TrailingWhitespaceTest {
     assert(xml"<foo> <bar/> </foo>" ≈ <foo> <bar/> </foo>)
   }
 
+  /*
```
Comment (nicolasstucki) [inline src/test/scala/dotty/xml/interpolator/TrailingWhitespaceTest.scala:33]: What happened with this test?

Comment (hamzaremmal) [reply src/test/scala/dotty/xml/interpolator/TrailingWhitespaceTest.scala:33]: If I recall correctly, this didn't work when I first launched the tests (before any changes). I'm not sure why and I will investigate later (hence commenting the test at the moment).

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**